src/utils/pgwire/src/pg_message.rs

```diff
@@ -43,11 +44,20 @@ pub struct FeQueryMessage {
 }
 
 impl FeQueryMessage {
-    pub fn get_sql(&self) -> &str {
+    pub fn get_sql(&self) -> Result<&str> {
         // Why there is a \0..
-        match std::str::from_utf8(&self.sql_bytes[..]) {
-            Ok(v) => v.trim_end_matches('\0'),
-            Err(e) => panic!("Invalid UTF-8 sequence: {}", e),
+        match CStr::from_bytes_with_nul(&self.sql_bytes) {
+            Ok(cstr) => match cstr.to_str() {
```

Comment:
This two match seems verbose. can we use `?`

Reply:
pub fn get_sql(&self) -> Result<&str> {
The result only support io error, so we need to repackage the error

Reply:
try map_err ? Or to_rw_results_with etc.